src/NEPTypes.jl

```diff
@@ -241,7 +241,18 @@ julia> compute_Mder(nep,1)-(A0+A1*exp(1))
                 end
             end
             ## Sum it up
-            Z=Z+nep.A[i]*(V*Fi);
+
+
+            VFi=V*Fi;
+            if (isa(nep.A[i],SubArray) && (eltype(nep.A[i]) != eltype(VFi)))
```

Comment:
Maybe put a "TODO" here. Just to remember that we should come back and check it every now and then.

Reply:
Done.